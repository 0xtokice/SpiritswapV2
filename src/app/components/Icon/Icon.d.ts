import { ReactNode } from 'react';

export interface StyledProps {
  size: number | string | 'big' | 'small' | 'normal';
  clickable?: boolean;
}

export interface Props extends StyledProps {
  icon: ReactNode;
}
