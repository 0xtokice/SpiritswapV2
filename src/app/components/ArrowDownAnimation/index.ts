export { default as ArrowDownAnimation } from './ArrowDown';
