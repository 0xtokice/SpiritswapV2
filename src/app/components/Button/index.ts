export type { Props as ButtonProps } from './Button.d';
export { default as Button } from './Button';
