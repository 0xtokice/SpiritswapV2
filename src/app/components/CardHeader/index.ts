export type { Props as CardHeaderProps } from './CardHeader.d';
export { default as CardHeader } from './CardHeader';
