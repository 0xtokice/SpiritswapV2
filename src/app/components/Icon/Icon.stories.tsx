import { ComponentStory, ComponentMeta } from '@storybook/react';

import Icon from './Icon';
import { Props } from './Icon.d';

import { ReactComponent as GhostIcon } from 'app/assets/images/ghost.svg';

// noinspection JSUnusedGlobalSymbols
export default {
  title: 'SpiritSwap/Components/Icon',
  component: Icon,
  args: {
    size: 'big',
    icon: <GhostIcon />,
  },
  argTypes: {
    size: {
      type: 'string',
    },
    icon: { control: { disable: true } },
  },
} as ComponentMeta<typeof Icon>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
// noinspection RequiredAttributes
const Template: ComponentStory<typeof Icon> = (args: Props) => (
  <Icon {...args} />
);

export const Custom = Template.bind({});
Custom.args = {
  size: '24px',
  icon: <GhostIcon />,
};

export const Big = Template.bind({});
Big.args = {
  size: 'big',
  icon: <GhostIcon />,
};

export const Small = Template.bind({});
Small.args = {
  size: 'small',
  icon: <GhostIcon />,
};
