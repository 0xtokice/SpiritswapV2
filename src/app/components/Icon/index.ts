export type { Props as IconProps } from './Icon.d';
export { default as Icon } from './Icon';
