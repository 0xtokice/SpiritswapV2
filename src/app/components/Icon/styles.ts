import styled from 'styled-components';
import { StyledProps } from './Icon.d';

export const StyledWrapper = styled.div<StyledProps>`
  display: flex;
  justify-content: center;
  align-items: center;

  ${({ size, theme }) => {
    let length;
    switch (size) {
      case 'big':
        length = theme.spacing.spacing07;
        break;
      case 'normal':
        length = theme.spacing.spacing06;
        break;
      case 'small':
        length = theme.spacing.spacing05;
        break;
      default:
        if (
          typeof size === 'number' ||
          (!isNaN(Number(size)) && !isNaN(parseFloat(size)))
        ) {
          length = `${size}px`;
        } else {
          length = size;
        }
    }
    return `
      &,
      & > div,
      & > svg,
      & > img {
        width: ${length};
        height: ${length};
      }
    `;
  }};

  display: flex;
  cursor: ${({ clickable = true }) => (clickable ? 'pointer' : 'default')};
`;
