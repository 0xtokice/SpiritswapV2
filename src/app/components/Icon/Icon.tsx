import { FC } from 'react';
import { Props } from './Icon.d';
import { StyledWrapper } from './styles';

const Icon: FC<Props> = ({ size, icon, ...props }: Props) => {
  return (
    <StyledWrapper size={size} {...props}>
      {icon}
    </StyledWrapper>
  );
};

export default Icon;
