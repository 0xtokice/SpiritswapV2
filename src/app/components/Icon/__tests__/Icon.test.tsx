import { render } from '@testing-library/react';
import { Icon, IconProps } from 'app/components/Icon';
import { ThemeWrapper } from 'app/components/shared/testing/ThemeWrapper';
import { ReactComponent as GhostIcon } from 'app/assets/images/ghost.svg';
import { spacing } from 'theme/base/spacing';

const defaultProps: IconProps = {
  icon: <GhostIcon />,
  size: 20,
};

const Component = (props: IconProps) => {
  return (
    <ThemeWrapper>
      <Icon {...props} />
    </ThemeWrapper>
  );
};

describe('Icon component', () => {
  test('renders without crashing', () => {
    const { container } = render(<Component {...defaultProps} />);
    expect(container.firstChild).toBeTruthy();
  });

  test('icon', () => {
    const { container } = render(<Component {...defaultProps} />);
    const icon = container.querySelectorAll('svg');
    expect(icon).toMatchSnapshot(`<svg>ghost.svg</svg>`);
  });

  test('size big', () => {
    const { container } = render(<Component {...defaultProps} size="big" />);
    expect(container.firstChild).toHaveStyle({
      width: spacing.spacing07,
      height: spacing.spacing07,
    });
  });

  test('size normal', () => {
    const { container } = render(<Component {...defaultProps} size="normal" />);
    expect(container.firstChild).toHaveStyle({
      width: spacing.spacing06,
      height: spacing.spacing06,
    });
  });

  test('size small', () => {
    const { container } = render(<Component {...defaultProps} size="small" />);
    expect(container.firstChild).toHaveStyle({
      width: spacing.spacing05,
      height: spacing.spacing05,
    });
  });

  test('size number', () => {
    const size = 40;
    const { container } = render(<Component {...defaultProps} size={size} />);
    expect(container.firstChild).toHaveStyle({
      width: `${size}px`,
      height: `${size}px`,
    });
  });
});
