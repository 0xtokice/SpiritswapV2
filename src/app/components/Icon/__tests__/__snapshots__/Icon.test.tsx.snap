// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component icon: <svg>ghost.svg</svg> 1`] = `
NodeList [
  <svg>
    ghost.svg
  </svg>,
]
`;
